src/libraries/System.Runtime.InteropServices.JavaScript/src/System/Runtime/InteropServices/JavaScript/JSSynchronizationContext.cs

```diff
@@ -50,9 +50,18 @@ public static unsafe JSSynchronizationContext InstallWebWorkerInterop(bool isMai
             ctx.previousSynchronizationContext = SynchronizationContext.Current;
             SynchronizationContext.SetSynchronizationContext(ctx);
 
-            if (JSProxyContext.ThreadBlockingMode == JSHostImplementation.JSThreadBlockingMode.NoBlockingWait)
+            if (!isMainThread)
             {
-                Thread.ThrowOnBlockingWaitOnJSInteropThread = true;
+                if (JSProxyContext.ThreadBlockingMode == JSHostImplementation.JSThreadBlockingMode.ThrowWhenBlockingWait)
```

Comment:
Why don't we unset `ThrowOnBlockingWaitOnJSInteropThread` and `WarnOnBlockingWaitOnJSInteropThread` the same as for main thread? Why do we need to unset them for main thread?

Reply:
Thanks!